modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,290 @@
  * Description: Uses WebP as the default format for new JPEG image uploads if the server supports it.
  * Experimental: No
  *
+ * @since   1.0.0
  * @package performance-lab
- * @since 1.0.0
  */
 
 /**
- * Filter the image editor default output format mapping.
+ * Hook called by `wp_generate_attachment_metadata` to create the `sources` property for every image
+ * size, the sources' property would create a new image size with all the mime types specified in
+ * `webp_uploads_valid_image_mime_types`. If the original image is one of the mimes from
+ * `webp_uploads_valid_image_mime_types` the image is just added to the `sources` property and  not
+ * created again. If the uploaded attachment is not a valid image this function does not alter the
+ * metadata of the attachment, on the other hand a `sources` property is added.
  *
- * For uploaded JPEG images, map the default output format to WebP.
+ * @since n.e.x.t
  *
- * @since 1.0.0
+ * @see   wp_generate_attachment_metadata
+ * @see   webp_uploads_valid_image_mime_types
  *
- * @param string $output_format The image editor default output format mapping.
- * @param string $filename      Path to the image.
- * @param string $mime_type     The source image mime type.
- * @return string The new output format mapping.
+ * @param array $metadata      An array with the metadata from this attachment.
+ * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+ *
+ * @return array An array with the updated structure for the metadata before is stored in the database.
```

Comment:
There shouldn't be blank lines between `@param` and `@return` in WP core.
```suggestion
 * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
 * @return array An array with the updated structure for the metadata before is stored in the database.
```